SCClassLibrary/Common/Math/Date.sc

```diff
@@ -1,13 +1,17 @@
-Date {
-	var <>year=2000, <>month=1, <>day=1, <>hour=0, <>minute=0, <>second=0, <>dayOfWeek=6,
-		<>rawSeconds=0;
+Date : Magnitude {
+	var <>year, <>month, <>day, <>hour, <>minute, <>second, <>dayOfWeek,
+		<>rawSeconds;
 
 	*getDate { ^this.localtime }
 	*localtime { ^this.new.localtime }
 	*gmtime { ^this.new.gmtime }
 	*new { arg year, month, day, hour, minute, second, dayOfWeek, rawSeconds;
-		^super.newCopyArgs(year, month, day, hour, minute, second, dayOfWeek,
-				rawSeconds);
+		var instance = super.newCopyArgs(year, month, day, hour, minute, second, dayOfWeek,
+			rawSeconds);
+		^instance.prInitialize;
+	}
+	*fromString { arg string, format;
+		^this.new.prFromString(string, format)
```

Comment:
Does this first resolve the date into current time and then sets if from a string? If so, then this would be one place to use the proposed `*basicNew` constructor